app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts

```diff
@@ -0,0 +1,84 @@
+import {module, IScope, extend} from 'angular';
+
+import {JSON_UTILITY_SERVICE, JsonUtilityService} from 'core/utils/json/json.utility.service';
+
+interface IViewState {
+  waitForCompletion?: boolean;
+  useStatusUrlFromLocationHeader?: boolean;
+}
+
+interface ICommand {
+  errorMessage?: string;
+  invalid?: boolean;
+  payloadJSON: string;
+}
+
+export class WebhookStage {
+  static get $inject() {
+    return ['stage', '$scope', 'jsonUtilityService'];
+  }
+
+  public command: ICommand;
+  public viewState: IViewState;
+  public methods: string[];
+
+  constructor(public stage: any,
+              $scope: IScope,
+              private jsonUtilityService: JsonUtilityService) {
+    this.methods = ['GET', 'POST', 'PUT'];
```

Comment:
These can be set directly on `methods` above (`public methods = ['GET', 'POST', 'PUT']`). Also..why not other HTTP methods?